alts/src/main/java/io/grpc/alts/AltsServerBuilder.java

```diff
@@ -172,40 +154,7 @@ public AltsServerBuilder intercept(ServerInterceptor interceptor) {
   /** {@inheritDoc} */
   @Override
   public Server build() {
-    if (!CheckGcpEnvironment.isOnGcp()) {
-      if (enableUntrustedAlts) {
-        logger.log(
-            Level.WARNING,
-            "Untrusted ALTS mode is enabled and we cannot guarantee the trustworthiness of the "
-                + "ALTS handshaker service");
-      } else {
-        Status status =
-            Status.INTERNAL.withDescription("ALTS is only allowed to run on Google Cloud Platform");
-        delegate.intercept(new FailingServerInterceptor(status));
-      }
-    }
-
-    delegate.protocolNegotiator(
-        AltsProtocolNegotiator.serverAltsProtocolNegotiator(handshakerChannelPool));
+    delegate.protocolNegotiator(credentialsBuilder.buildProtocolNegotiator());
```

Comment:
This changes the behavior, right? Instead of failing each call now we fail protocol negotiation i.e. just once at the beginning of channel creation. I am okay with that but wondering about backward compatibility for error behavior.

Reply:
This was the same approach taken on client-side. RPCs receive the same error with this approach. The only difference is things like name resolution still occur.

Reply:
And I remember commenting on the client-side but forgot what the outcome was. SG.

Reply:
Hmm... Actually, this causes things to react differently on server-side, as the client won't see this and it probably won't be logged. So this is sort of broken. But the server interceptor was probably never even called earlier, as the handshake itself would fail, so it seems like it was already broken.